packages/trimerge-sync-basic-server/src/lib/SqliteDocStore.ts

```diff
@@ -98,19 +109,53 @@ export class SqliteDocStore implements DocStore {
         VALUES (@ref, @remoteSyncId, @remoteSyncIndex, @userId, @clientId, @baseRef, @mergeRef, @mergeBaseRef, @delta, @editMetadata)
         ON CONFLICT DO NOTHING`,
     );
-    const refs: string[] = [];
+    const knownRefs = new Set(this.seenRefs);
+    for (const { ref } of nodes) {
+      knownRefs.add(ref);
+    }
+    const refs = new Set<string>();
+    const refErrors: AckRefErrors = {};
+    function invalidParentRef(
+      node: DiffNode<unknown, unknown>,
+      key: keyof Pick<
+        DiffNode<unknown, unknown>,
+        'baseRef' | 'mergeRef' | 'mergeBaseRef'
+      >,
```

Comment:
Would this be sufficient/simpler:
```suggestion
      key: 'baseRef' | 'mergeRef' | 'mergeBaseRef',
```

Reply:
oh yea! I had done it for type safety, but it looks `node[key]` gives me the equivalent check